engines/router/missionctl/server/upi/server.go

```diff
@@ -38,17 +37,18 @@ type Server struct {
 	upiv1.UnimplementedUniversalPredictionServiceServer
 
 	missionControl missionctl.MissionControlUPI
+	resultLogger   *resultlog.UPIResultLogger
 }
 
-func NewUPIServer(mc missionctl.MissionControlUPI) *Server {
+func NewUPIServer(mc missionctl.MissionControlUPI, rl *resultlog.UPIResultLogger) *Server {
 	return &Server{
 		missionControl: mc,
+		resultLogger:   rl,
 	}
 }
 
 func (us *Server) Run(listener net.Listener) {
 	s := grpc.NewServer()
-	//TODO: the unmarshalling can be done more efficiently by using partial deserialization
```

Comment:
initially optimization could had been done to speed up calls, however it would be required for logging purposes, hence there is no more need to explore this